hatchways/team-bluejay
# BE: Moshe store favorite cuisines #18

Ticket #18 

Description:

- Added many to many relationship between users and their favorite cuisines
- The tables are Users, Favorite_Cuisines, and a Cuisines table
- Added a seeder file `db_seeder.py` to add cuisines to database
- Added flask-migrate to run a script to update database tables when Models change (we will no longer have to delete database and reinitialize it whenever we make a table change)
- added `python-dotenv` to allow easily setting up variables in .env file

Testing:

- Change your .env to below variables
- Run `pipenv install`
- Run `python db_seeder.py` to add cuisines to database
- Create a new user and then use Postman PUT "Update User" with `"cuisines": [{"id": 5}, {"id": 2}]` in json body to add favorite cuisines for user.   Check that returned user object in postman has cuisines. Check in Postgres GUI under "favorite_cuisines" that there is a relationship between user_id and cuisine_id
- Update user again with `"cuisines": [{"id": 1}]`. Make sure user has relationship with correct cuisine.
- Run Postman GET "Get All Cuisines" and ensure you get all cuisines


# .env

TEAM_NAME="Shums, Kareem, Gabriel, Moshe"
FLASK_ENV="development"

MOSHES_GOOGLE_API_KEY = 'AIzaSyAN3q6E8D80h38QQsiSWQ5keW6YSRMPqYc'

DB_USER = 'postgres'
DB_PASSWORD = '123456'
DB_URL = 'localhost:5432'
DB_NAME = 'team-bluejay'
